projects/packages/sync/src/modules/class-module.php

```diff
@@ -346,6 +346,15 @@ public function send_full_sync_actions( $config, $status, $send_until ) {
 
 			$result = $this->send_action( 'jetpack_full_sync_' . $this->name(), array( $objects, $status['last_sent'] ) );
 
+			if ( is_wp_error( $result ) && 'full_sync_max_upload_exceeded' === $result->get_error_code() && $limits['chunk_size'] > 1 ) {
+				// If we hit the max upload size, we need to reduce the chunk size and try again.
+				// Also resetting send_until and chunks_sent to only send one chunk with the new chunk_size.
+				$limits['chunk_size'] = (int) ( $limits['chunk_size'] / 2 );
+				$chunks_sent          = $limits['max_chunks'] - 1;
+				$send_until           = microtime( true ) + Settings::get_setting( 'full_sync_send_duration' );
```

Comment:
Not sure if there is a better way to "reset" `$send_until`

Reply:
I wonder if we actually need to reset `$send_until`. It would make more sense to me to keep it as is since the Full Sync duration should be independent of how we chunk the items.

Reply:
From what I've gathered this `$send_until` is for the current `send` request in https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/modules/class-full-sync-immediately.php#L381 for each module, and it is just set to 9 microseconds after. So in our case in the next iteration of the while, the check below in LINE 343 will make us return early.
```
if ( $chunks_sent++ === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
	return $status;	
}
```
Not sure how are we getting past this for sync objects with `$limits['max_chunks']` greater than 1 though.